core/src/apps/cardano/sign_tx.py

```diff
@@ -4,32 +4,37 @@
 from trezor.crypto import base58, hashlib
 from trezor.crypto.curve import ed25519
 from trezor.messages.CardanoSignedTx import CardanoSignedTx
-from trezor.messages.CardanoTxAck import CardanoTxAck
-from trezor.messages.CardanoTxRequest import CardanoTxRequest
 
 from apps.cardano import CURVE, seed
 from apps.cardano.address import (
     derive_address_and_node,
     is_safe_output_address,
     validate_full_path,
 )
-from apps.cardano.layout import confirm_sending, confirm_transaction, progress
+from apps.cardano.layout import confirm_sending, confirm_transaction
 from apps.common import cbor
 from apps.common.paths import validate_path
 from apps.common.seed import remove_ed25519_prefix
 
+if False:
+    from typing import Dict, List, Tuple
+    from trezor.messages.CardanoSignTx import CardanoSignTx
+    from trezor.messages.CardanoTxInputType import CardanoTxInputType
+    from trezor.messages.CardanoTxOutputType import CardanoTxOutputType
+
 # the maximum allowed change address.  this should be large enough for normal
 # use and still allow to quickly brute-force the correct bip32 path
 MAX_CHANGE_ADDRESS_INDEX = const(1000000)
 ACCOUNT_PREFIX_DEPTH = const(2)
 
-KNOWN_PROTOCOL_MAGICS = {764824073: "Mainnet", 1097911063: "Testnet"}
+LOVELACE_MAX_SUPPLY = 45_000_000_000 * 1_000_000
 
 
-# we consider addresses from the external chain as possible change addresses as well
-def is_change(output, inputs):
+# addresses from the same account as inputs should be hidden
+def should_hide_output(output: List[int], inputs: List[CardanoTxInputType]) -> bool:
     for input in inputs:
         inp = input.address_n
+        print(output[:ACCOUNT_PREFIX_DEPTH])
```

Comment:
forgotten print

Reply:
Omg, and on the same spot. Sorry :disappointed: Removed.